spring-web/src/test/java/org/springframework/web/service/invoker/HttpMethodArgumentResolverTests.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2002-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.web.service.invoker;
+
+import java.util.Collections;
+
+import org.junit.jupiter.api.Test;
+import reactor.core.publisher.Mono;
+import reactor.test.StepVerifier;
+
+import org.springframework.http.HttpMethod;
+import org.springframework.web.service.annotation.GetRequest;
+import org.springframework.web.service.annotation.HttpRequest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.web.service.invoker.HttpServiceMethodTestUtils.createService;
+
+/**
+ * Tests for {@link HttpMethodArgumentResolver}.
+ *
+ * @author Olga Maciaszek-Sharma
+ */
+class HttpMethodArgumentResolverTests {
+
+	private final TestHttpClientAdapter clientAdapter = new TestHttpClientAdapter();
+	private final Service service = createService(Service.class, clientAdapter, Collections.singletonList(new HttpMethodArgumentResolver()));
+
+	@Test
+	void shouldResolveRequestMethodFromArgument() {
+		Mono<Void> execution = service.execute(HttpMethod.GET);
+
+		StepVerifier.create(execution).verifyComplete();
+		assertThat(clientAdapter.getRequestDefinition()
+				.getHttpMethod()).isEqualTo(HttpMethod.GET);
+	}
+
+	@Test
+	void shouldIgnoreArgumentsNotMatchingType() {
+		Mono<Void> execution = service.execute("test");
+
+		StepVerifier.create(execution).verifyComplete();
+		assertThat(clientAdapter.getRequestDefinition()
+				.getHttpMethod()).isNull();
+	}
+
+	@Test
+	void shouldOverrideMethodAnnotationWithMethodArgument() {
+		Mono<Void> execution = service.executeGet(HttpMethod.POST);
+
+		StepVerifier.create(execution).verifyComplete();
+		assertThat(clientAdapter.getRequestDefinition()
+				.getHttpMethod()).isEqualTo(HttpMethod.POST);
+	}
+
+	@Test
+	void shouldOverrideMethodWithSubsequentMethodArgument() {
```

Comment:
Maybe we don't need to explicitly have a test for this. In other words, we don't officially support endorse, or expect this. It happens to work as the test shows, but in practice I don't see a reason to do this in an, and while we could reject, I'm not sure it's worth the trouble.

Reply:
Ok, will remove it then.